drivers/ieee802154/ieee802154_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,735 @@
+{
+	ARG_UNUSED(dev);
+	return 0;
+}
+
+static int ieee802154_cc13xx_cc26xx_stop(struct device *dev)
+{
+	ARG_UNUSED(dev);
+	RFCDoorbellSendTo(CMDR_DIR_CMD(CMD_STOP));
+	return 0;
+}
+
+static void ieee802154_cc13xx_cc26xx_cpe0_isr(void *arg)
+{
+	struct ieee802154_cc13xx_cc26xx_data *drv_data = get_dev_data(arg);
+
+	u32_t flags = RFCCpeIntGetAndClear(IRQ_RX_ENTRY_DONE | IRQ_TX_DONE);
+
+	if (flags & IRQ_RX_ENTRY_DONE) {
+		k_sem_give(&drv_data->rx_done);
+	}
+
+	if (flags & IRQ_TX_DONE) {
+		k_sem_give(&drv_data->tx_done);
+	}
+}
+
+static void ieee802154_cc13xx_cc26xx_cpe1_isr(void *arg)
+{
+	ARG_UNUSED(arg);
+
+	u32_t flags =
+		RFCCpeIntGetAndClear(IRQ_RX_BUF_FULL | IRQ_INTERNAL_ERROR);
+
+	if (flags & IRQ_RX_BUF_FULL) {
+		LOG_WRN("Receive buffer full");
+	}
+
+	if (flags & IRQ_INTERNAL_ERROR) {
+		LOG_ERR("Internal error");
+	}
+}
+
+static void ieee802154_cc13xx_cc26xx_data_init(struct device *dev)
+{
+	struct ieee802154_cc13xx_cc26xx_data *drv_data = get_dev_data(dev);
+
+	u8_t *mac = (u8_t *)(FCFG1_BASE + FCFG1_O_MAC_15_4_0);
+	if (sys_read32(CCFG_BASE + CCFG_O_IEEE_MAC_0) != 0xFFFFFFFF &&
+	    sys_read32(CCFG_BASE + CCFG_O_IEEE_MAC_1) != 0xFFFFFFFF) {
+		mac = (u8_t *)(CCFG_BASE + CCFG_O_IEEE_MAC_0);
+	}
+	memcpy(&drv_data->ext_addr, mac, sizeof(drv_data->ext_addr));
+	drv_data->short_addr = 0x0000;
+	drv_data->pan_id = 0x0000;
+
+	/* Setup circular RX queue (TRM 25.3.2.7) */
+	memset(&drv_data->rx_entry[0], 0, sizeof(drv_data->rx_entry[0]));
+	memset(&drv_data->rx_entry[1], 0, sizeof(drv_data->rx_entry[1]));
+
+	drv_data->rx_entry[0].pNextEntry = (u8_t *)&drv_data->rx_entry[1];
+	drv_data->rx_entry[0].config.type = DATA_ENTRY_TYPE_PTR;
+	drv_data->rx_entry[0].config.lenSz = 1;
+	drv_data->rx_entry[0].length = sizeof(drv_data->rx_data[0]);
+	drv_data->rx_entry[0].pData = drv_data->rx_data[0];
+
+	drv_data->rx_entry[1].pNextEntry = (u8_t *)&drv_data->rx_entry[0];
+	drv_data->rx_entry[1].config.type = DATA_ENTRY_TYPE_PTR;
+	drv_data->rx_entry[1].config.lenSz = 1;
+	drv_data->rx_entry[1].length = sizeof(drv_data->rx_data[1]);
+	drv_data->rx_entry[1].pData = drv_data->rx_data[1];
+
+	drv_data->rx_queue.pCurrEntry = (u8_t *)&drv_data->rx_entry[0];
+	drv_data->rx_queue.pLastEntry = NULL;
+
+	k_sem_init(&drv_data->tx_done, 0, 1);
+	k_sem_init(&drv_data->rx_done, 0, 1);
+	k_sem_init(&drv_data->ack, 0, 1);
+
+	k_thread_create(&drv_data->rx_thread, drv_data->rx_stack,
+			K_THREAD_STACK_SIZEOF(drv_data->rx_stack),
+			ieee802154_cc13xx_cc26xx_rx, dev, NULL, NULL,
+			K_PRIO_COOP(2), 0, K_NO_WAIT);
+}
+
+static int ieee802154_cc13xx_cc26xx_init(struct device *dev)
+{
+	struct ieee802154_cc13xx_cc26xx_data *drv_data = get_dev_data(dev);
+
+	/* Apply RF patches */
+#ifdef CONFIG_SOC_SERIES_CC13X2_CC26X2
+	rf_patch_cpe_ieee_802_15_4();
+#endif /* CONFIG_SOC_SERIES_CC13X2_CC26X2 */
+
+	/* Set high frequency clock source (TRM 7.5.1.1) */
+	bool set_osc_hf = OSCClockSourceGet(OSC_SRC_CLK_HF) != OSC_XOSC_HF;
```

Comment:
Always declare a variable in the beginning of the code block it belongs. apply that everywhere

Reply:
I went through and applied this pattern everywhere. Let me know if I missed something.